crates/topos-tce/src/app_context/network.rs

```diff
@@ -20,69 +24,99 @@ impl AppContext {
+                            if let hash_map::Entry::Vacant(entry) =
+                                self.delivery_latency.entry(cert.id)
                             {
-                                error!(
-                                    "Unable to send broadcast_new_certificate command, Receiver \
-                                     was dropped"
-                                );
+                                entry.insert(CERTIFICATE_DELIVERY_LATENCY.start_timer());
                             }
-                        });
+
+                            spawn(async move {
+                                info!("Send certificate {} to be broadcast", cert.id);
+                                if channel
+                                    .send(DoubleEchoCommand::Broadcast {
+                                        cert,
+                                        need_gossip: false,
+                                    })
+                                    .await
+                                    .is_err()
+                                {
+                                    error!(
+                                        "Unable to send broadcast_new_certificate command, \
+                                         Receiver was dropped"
+                                    );
+                                }
+                            });
+                        }
                     }
-                    TceCommands::OnEcho {
-                        certificate_id,
-                        signature,
-                        validator_id,
-                    } => {
+                    double_echo_request::Request::Echo(EchoRequest {
+                        certificate: Some(certificate_id),
+                        signature: Some(signature),
+                        validator_id: Some(validator_id),
+                    }) => {
                         let channel = self.tce_cli.get_double_echo_channel();
                         spawn(async move {
-                            if let Err(e) = channel
-                                .send(DoubleEchoCommand::Echo {
-                                    signature,
-                                    certificate_id,
-                                    validator_id,
-                                })
-                                .await
+                            let certificate_id = certificate_id.try_into().map_err(|e| {
+                                error!("Invalid certificate id, could not send Echo message: {e}");
+                                e
+                            });
+                            let validator_id = validator_id.try_into().map_err(|e| {
+                                error!("Invalid validator id, could not send Echo message: {e}");
+                                e
+                            });
+                            if let (Ok(certificate_id), Ok(validator_id)) =
```

Comment:
Same can we have a log to see errors?

Reply:
Errors are printed above while converting `certificate_id` and `validator_id`. I'll add one more summary error.